packages/iTwinUI-react/src/core/Table/hooks/useStickyColumns.tsx

```diff
@@ -0,0 +1,98 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import {
+  actions,
+  ActionType,
+  Hooks,
+  TableInstance,
+  TableState,
+} from 'react-table';
+
+actions.setScrolledLeft = 'setScrolledLeft';
+actions.setScrolledRight = 'setScrolledRight';
+
+export const useStickyColumns = <T extends Record<string, unknown>>(
+  hooks: Hooks<T>,
+) => {
+  hooks.stateReducers.push(reducer);
+  hooks.useInstance.push(useInstance);
+};
+
+const reducer = <T extends Record<string, unknown>>(
+  newState: TableState<T>,
+  action: ActionType,
+): TableState<T> => {
+  if (action.type === actions.init) {
+    return {
+      ...newState,
+      sticky: {},
+    };
+  }
+
+  if (
+    action.type === actions.setScrolledLeft &&
+    newState.sticky?.isScrolledToLeft !== action.value // Prevents unnecessary re-render
+  ) {
+    return {
+      ...newState,
+      sticky: {
+        ...newState.sticky,
+        isScrolledToLeft: action.value,
+      },
+    };
+  }
+
+  if (
+    action.type === actions.setScrolledRight &&
+    newState.sticky?.isScrolledToRight !== action.value // Prevents unnecessary re-render
+  ) {
+    return {
+      ...newState,
+      sticky: {
+        ...newState.sticky,
+        isScrolledToRight: action.value,
+      },
+    };
+  }
+
+  return newState;
+};
+
+const useInstance = <T extends Record<string, unknown>>(
+  instance: TableInstance<T>,
+) => {
+  const { flatHeaders } = instance;
+
+  // Edge case. Saving original sticky state in case sticky columns are reordered.
+  flatHeaders.forEach((header) => {
+    if (!header.originalSticky) {
+      header.originalSticky = header.sticky ?? 'none';
```

Comment:
why new value is introduced? `sticky` values were `right`, `left` or `undefined`

Reply:
I need to know what columns originally were not sticky so I need some third type that tells that column was not sticky. I can't use `undefined` for that because it will be overwritten by the same if.